### PR TITLE
Deprecate a bunch of engine things

### PR DESCRIPTION
**Pull request by chrisnojima**
This PR does some long standing cleanup to the engine layer. 
The PRs before this switch from the registerIncomingCall flow to just emitting actions so we can handle things w/ our sagas in a similar way and reduce a bunch of boiler plate
This PR handles the rarer case where we want to handle an incoming call with an action, yet we still want to handle the response ourselves.
The simplest way to do this was to register the method which tells the engine just to not auto respond
This does touch some super old rarely used flow so lets merge after the release

- [ ]  after release

### CODE REVIEW

File: protocol/bin/flow.js [hunk 1]
```diff
@@ -394,8 +394,14 @@ function writeActions() {
               .split('.')
               .map((p, idx) => (idx ? capitalize(p) : p))
               .join('')
+
+            let response = ''
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: plumb through response object

File: shared/actions/config/index.js [hunk 2]
```diff
@@ -118,15 +118,6 @@ function* loadDaemonBootstrapStatus(state, action) {
   }
 }
 
-let dispatchSetupEngineListenersOnce = false
```
Comment (chrisnojima) [inline shared/actions/config/index.js]: we don't dispatch this anymore

File: shared/actions/fs/platform-specific.desktop.js [hunk 3]
```diff
@@ -356,8 +356,8 @@ function* platformSpecificSaga(): Saga.SagaGenerator<any, any> {
     FsGen.openPathInSystemFileManager,
     openPathInSystemFileManager
   )
-  yield* Saga.chainGenerator<ConfigGen.SetupEngineListenersPayload | FsGen.FuseStatusPayload>(
-    [ConfigGen.setupEngineListeners, FsGen.fuseStatus],
+  yield* Saga.chainGenerator<ConfigGen.InstallerRanPayload | FsGen.FuseStatusPayload>(
```
Comment (chrisnojima) [inline shared/actions/fs/platform-specific.desktop.js]: this is a close proxy for the behavior before

File: shared/actions/pinentry.js [hunk 4]
```diff
@@ -1,21 +1,16 @@
...
 import * as PinentryGen from '../actions/pinentry-gen'
 import * as Saga from '../util/saga'
 import * as I from 'immutable'
 import * as RPCTypes from '../constants/types/rpc-gen'
-import engine from '../engine'
+import {getEngine} from '../engine/require'
 
-// We keep track of sessionID to response objects since this is initiated by the daemon
-type IncomingErrorCallback = (?{code?: number, desc?: string}) => void
-const sessionIDToResponse: {
```
Comment (chrisnojima) [inline shared/actions/pinentry.js]: spoke to @maxtaco about this. this likely never worked. so lets simplify and just keep one response. in the future if we want this to be smarter its likely easier for the GUI to make a call to the daemon registering itself with an engineSaga and just getting pinenetry requests with a 'real id' and not this session id

File: shared/actions/pinentry.js [hunk 5]
```diff
@@ -60,50 +49,38 @@ const onNewPinentry = (_, action) =>
   })
 
 const onSubmit = (_, action) => {
-  const {sessionID, passphrase} = action.payload
-  _respond(sessionID, {passphrase})
+  const {passphrase} = action.payload
+  if (_response) {
+    // $FlowIssue flow is correct in that we need store secret but we weren't sending it before and i don't want to change any existing behavior
+    _response.result({passphrase})
```
Comment (chrisnojima) [inline shared/actions/pinentry.js]: flow will infer _response correctly now and it correctly points out we're missing a param. i didn't want to break anything so i left it as-is

File: shared/actions/platform-specific/index.desktop.js [hunk 6]
```diff
@@ -332,18 +329,6 @@ export function* platformConfigSaga(): Saga.SagaGenerator<any, any> {
   )
   yield* Saga.chainAction<ConfigGen.ShowMainPayload>(ConfigGen.showMain, showMainWindow)
   yield* Saga.chainAction<ConfigGen.DumpLogsPayload>(ConfigGen.dumpLogs, dumpLogs)
-  yield* Saga.chainGenerator<ConfigGen.SetupEngineListenersPayload>(
```
Comment (chrisnojima) [inline shared/actions/platform-specific/index.desktop.js]: there's no value in waiting for this action so just spawn it

File: shared/actions/platform-specific/index.desktop.js [hunk 7]
```diff
@@ -361,6 +346,11 @@ export function* platformConfigSaga(): Saga.SagaGenerator<any, any> {
     EngineGen.keybase1NotifySessionClientOutOfDate,
     onOutOfDate
   )
+  getEngine().registerCustomResponse('keybase.1.logsend.prepareLogsend')
```
Comment (chrisnojima) [inline shared/actions/platform-specific/index.desktop.js]: this is how we tell the engine that we intend on responding to these actions ourselves

File: shared/actions/tracker.js [hunk 8]
```diff
@@ -286,7 +285,8 @@ function _serverCallMap(
 
         // cleanup bookkeeping
         delete sessionIDToUsername[sessionID]
-        engine().cancelSession(sessionID)
+        // $FlowIssue exist but this is the only thing that uses it and this is going away
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: this entire file is going away after the next release

**Comment by chrisnojima** [comment]
@keybase/react-hackers @jzila @songgao

File: shared/actions/fs/platform-specific.desktop.js (hunk 3 above)
Comment (songgao) [reply shared/actions/fs/platform-specific.desktop.js]: Does this always happen regardless of user settings (fuse driver, autostart, etc.)?

Comment (chrisnojima) [reply shared/actions/fs/platform-specific.desktop.js]: this matches the logic before. before config saga would listen for InstallerRan and dispatch setupEngineListeners

Comment (songgao) [reply shared/actions/fs/platform-specific.desktop.js]: Got it; sounds good!

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Pull request merged**